controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -273,6 +273,9 @@ type KubeadmControlPlaneStatus struct {
 	// Ready denotes that the KubeadmControlPlane API Server is ready to
 	// receive requests.
 	// +optional
+	// NOTE: this field is part of the Cluster API contract and it is used to orchestrate provisioning.
+	// The value of this field is never updated after provisioning is completed. Please use conditions
+	// to check the operational state of the control plane.
```

Comment:
i have not context. leaving to active maintainer feedback.

Reply:
In general the comment is fine and there was a lengthy discussion about it on the issue.

I would like to update `Ready denotes that the KubeadmControlPlane API Server is ready to receive requests.` as well though. In my opinion it is highly misleading and in conflict with the added note. Let's replace that one with:

```go
	// Ready denotes that the KubeadmControlPlane API Server became ready during initial provisioning to
	// receive requests.
```